### ISSUE TITLE
inconsistent results of ReMixMatch

### ISSUE THREAD

**Issue by bluelg** [bbuing9/DARP#3]
--

**Comment by bluelg** [comment]
Thanks for your nice work!

There is one question when I run your code of ReMixMatch and ReMixMatch + DARP following your suggestion.

ReMixMatch: python train_remix.py --align --gpu 0 --ratio 2 --num_max 1500 --imb_ratio_l 150 --imb_ratio_u 150 --epoch 500 --val-iteration 500 --out cifar10@N_1500_r_150_remix

ReMixMatch+DARP: python train_remix.py --darp --align --alpha 2 --warm 200 --gpu 0 --ratio 2 --num_max 1500 --imb_ratio_l 150 --imb_ratio_u 150 --epoch 500 --val-iteration 500 --out cifar10@N_1500_r_150_remix_darp

After running the evaluation code through
python eval.py --pre_trained ./cifar10@N_1500_r_150_remix
python eval.py --pre_trained ./cifar10@N_1500_r_150_remix_darp

I find that inconsistent results of these two methods with those reported in Table 1.

./cifar10@N_1500_r_150_remix_darp
Average bACC of last 20 epochs : 71.10900000000001
Average GM of last 20 epochs : 0.6506487500000001

./cifar10@N_1500_r_150_remix
Average bACC of last 20 epochs : 71.55999999999997
Average GM of last 20 epochs : 0.6586442499999998

It seems that DARP cannot enhance ReMixMatch, could you help explain that?

**Comment by bbuing9** [comment]
Hi,

One possible reason is the randomness from each random seed. Due to this reason, even for the multiple trials of the same method (ReMixMatch or DARP), the results can be different among them. Hence, for clearly verifying the gain from DARP, it would be better to 1) use the same model and random seed for ReMixMatch and DARP in case of a single run; Namely, applying DARP to the intermediate checkpoints generated from the running of ReMixMatch ( saved as model_200.pth.tar ). Or 2) run multiple trials for each method to reduce the effect of randomness.

Additionally, as DARP is more effective under a large distribution mismatch, it would be easier to verify the gain of DARP by conducting the experiments under a more challenging scenario; imb_ratio_l = 100 & imb_ratio_u = 1 as described in "run.sh" (Table2)

**Comment by JoyHuYY1412** [comment]
> Hi,
> 
> One possible reason is the randomness from each random seed. Due to this reason, even for the multiple trials of the same method (ReMixMatch or DARP), the results can be different among them. Hence, for clearly verifying the gain from DARP, it would be better to 1) use the same model and random seed for ReMixMatch and DARP in case of a single run; Namely, applying DARP to the intermediate checkpoints generated from the running of ReMixMatch ( saved as model_200.pth.tar ). Or 2) run multiple trials for each method to reduce the effect of randomness.
> 
> Additionally, as DARP is more effective under a large distribution mismatch, it would be easier to verify the gain of DARP by conducting the experiments under a more challenging scenario; imb_ratio_l = 100 & imb_ratio_u = 1 as described in "run.sh" (Table2)

Hi @bbuing9, if I want to run some cases of different imb_ratios, I need to run train_base_estim.py to get the predicted label distribution first, right? 
And could you please tell me what the val_loader is for?
https://github.com/bbuing9/DARP/blob/32fc4b0df234b07f715ff4d29ce4bb39f8d4d8fb/train_base_estim.py#L93-L98
Does the size of val_set influence the estimation? 

And why do you exclude those validation data in the training set as following? I think including them in training may be better and necessary when the labeled data is scarce.
https://github.com/bbuing9/DARP/blob/32fc4b0df234b07f715ff4d29ce4bb39f8d4d8fb/dataset/base_cifar10_split.py#L49-L53
